crates/driver/src/infra/api/routes/solve/dto/solved.rs

```diff
@@ -0,0 +1,48 @@
+use {
+    crate::{
+        domain::{competition, eth},
+        infra::Solver,
+        util::serialize,
+    },
+    serde::Serialize,
+    serde_with::serde_as,
+};
+
+impl Solved {
+    pub fn new(solved: competition::Solved, solver: &Solver) -> Self {
+        let solution = Solution::new(0, solved, solver);
+        Self {
+            solutions: vec![solution],
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Solved {
+    solutions: Vec<Solution>,
+}
+
+impl Solution {
+    pub fn new(solution_id: u64, solved: competition::Solved, solver: &Solver) -> Self {
+        Self {
+            solution_id,
+            score: solved.score.into(),
+            submission_address: solver.address().into(),
+        }
+    }
+}
+
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Solution {
+    /// Unique ID of the solution, used to identify it in subsequent requests
+    /// (reveal, settle).
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    solution_id: u64,
```

Comment:
I really think this should be a new type (see how we do auction and order IDs)

Reply:
Something like `SolutionId(u64)` or?
Not following since the auction id is also [simple type](https://github.com/cowprotocol/services/blob/main/crates/driver/src/infra/api/routes/solve/dto/auction.rs#L174).

Reply:
Ok, now I see `auction::Id` but that is in the `driver` domain which is out of scope of this PR. This PR only changes the `autopilot/driver` interface and makes minimal changes in `driver` to support the interface change.

Reply:
Sorry, I thought this was the domain and not the DTO. You are right that we don't need new-types there.